.github/workflows/hosted-execution-no-docker-hub.yml

```diff
@@ -0,0 +1,57 @@
+name: Hosted Execution sans Docker Image
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+
+jobs:
+  retain-files:
+    name: Upload Test Files to Sauce Storage
+    runs-on: ubuntu-latest
+    env:
+      SAUCE_USERNAME: ${{ secrets.SAUCE_USERNAME }}
+      SAUCE_ACCESS_KEY: ${{ secrets.SAUCE_ACCESS_KEY }}
+      SAUCE_REGION: us-west-1
+    outputs:
+      storage_id: ${{ env.STORAGE_ID }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install Saucectl
+        run: npm install -g saucectl
+      - name: Archive test files
+        run: zip -r files.zip . -x "**test-apps/*" ".git/*" "**node_modules/*"
+      - name: Push artifacts to Sauce Storage
+        run: saucectl storage upload files.zip &> sauce-storage-output.log
```

Comment:
Btw. if you have `jq` installed, you could also do:
`STORAGE_ID=$(saucectl storage upload files.zip -o json | jq .id)`

Reply:
Oh sweet thanks, I didn't know saucectl supported JSON output, that's much better.